pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -256,20 +273,93 @@ func parseMstconfigOutput(mstOutput string, attributes []string) (fwCurrent, fwN
 	return
 }
 
-func isFirstPort(pciAddress string) bool {
-	glog.Infof("mellanox-plugin isFirstPort(): device %s", pciAddress)
-	return pciAddress[len(pciAddress)-1] == '0'
+func getPciAddressPrefix(pciAddress string) string {
+	return pciAddress[:len(pciAddress)-1]
 }
 
-func isMlnxNicAndInNode(pciAddress string, state *sriovnetworkv1.SriovNetworkNodeState) bool {
-	glog.Infof("mellanox-plugin isMlnxNicAndInNode(): device %s", pciAddress)
-	for _, iface := range state.Status.Interfaces {
-		if iface.PciAddress == pciAddress {
-			if iface.Vendor == MellanoxVendorId {
-				return true
-			}
-			return false
+func isDualPort(pciAddress string) bool {
+	glog.Infof("mellanox-plugin isDualPort(): pciAddress %s", pciAddress)
+	pciAddressPrefix := getPciAddressPrefix(pciAddress)
+	isDual := len(mellanoxNicsStatus[pciAddressPrefix]) > 1
```

Comment:
Is checking the PCI address sufficient for judging the dual-port NIC?

Reply:
yes